projects/continuous_learning/fb_sparsenet.py

```diff
@@ -0,0 +1,272 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2019, Numenta, Inc.  Unless you have an agreement
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see http://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+
+import numpy as np
+import torch
+from torch import nn
+
+from nupic.research.frameworks.pytorch.modules import KWinners2dLocal
+from nupic.research.frameworks.pytorch.modules.consolidated_sparse_weights import (
+    ConsolidatedSparseWeights,
+    ConsolidatedSparseWeights2D,
+)
+from nupic.torch.modules import (
+    Flatten,
+    KWinners,
+    KWinners2d,
+    SparseWeights,
+    SparseWeights2d,
+)
+
+
+class FBNet(nn.Module):
+    def __init__(self,
```

Comment:
Would benefit from a docstring. If it just experimental code, doesn't have to something fancy, but 2 or 3 lines explaining inputs, outputs and what its doing at a high level helps a reviewer or a future version of you going back to this code.